build.rs

```diff
@@ -1,191 +1,393 @@
+        "DYNAMIC" => { freeimage_options.push(format!("-DUSE_FREEIMAGE_STATIC:BOOL={}", "OFF")); },
+        _ => fail("Invalid freeimage build option set"),
+    };
+
     let mut cmake_cmd = Command::new("cmake");
     cmake_cmd.current_dir(&build_dir);
+
     run(cmake_cmd.arg("..")
-        .arg(format!("-DCMAKE_BUILD_TYPE={}", conf.release_type))
-        .arg(format!("-DBUILD_CUDA={}", conf.build_cuda))
-        .arg(format!("-DBUILD_OPENCL={}", conf.build_opencl))
-        .arg(format!("-DBUILD_EXAMPLES={}", conf.build_examples))
-        .arg(format!("-DBUILD_TEST={}", conf.build_test))
-        .arg(format!("-DBUILD_CPU={}", conf.build_cpu)), "cmake");
+        .args(&[format!("-DCMAKE_BUILD_TYPE:STRING={}", conf.build_type),
+                format!("-DBUILD_CPU:BOOL={}", conf.build_cpu),
+                format!("-DBUILD_CUDA:BOOL={}", conf.build_cuda),
+                format!("-DBUILD_OPENCL:BOOL={}", conf.build_opencl),
+                format!("-DBUILD_EXAMPLES:BOOL={}", conf.build_examples),
+                format!("-DBUILD_TEST:BOOL={}", conf.build_test),
+                format!("-DCMAKE_INSTALL_PREFIX:STRING={}", "package")])
+        .args(&freeimage_options)
+        .args(&blas_options)
+        .args(&lapack_options)
+        .args(&graphics_options)
+        , "cmake");
 
     // run make
-    let mut make_cmd = Command::new("make");
+    let mut make_cmd= Command::new("make");
     make_cmd.current_dir(&build_dir);
-    run(make_cmd.arg(conf.make_flags.to_owned()), "make");
-  }
-
-  // build correct backend
-  let (backends, backend_dirs) = blob_backends(&conf, &build_dir);
-  for backend in backends.iter() {
-    println!("cargo:rustc-link-lib=dylib={}", backend);
-  }
-
-  for backend_dir in backend_dirs.iter() {
-    println!("cargo:rustc-link-search=native={}", backend_dir);
-  }
-
-  if conf.use_lib {
-    arrayfire_dir = PathBuf::from(conf.inc_dir);
-  }else{
-    arrayfire_dir = arrayfire_dir.join("include");
-  }
-  
-  build_bindings("arrayfire", &src_dir, &arrayfire_dir);
+    run(make_cmd
+        .arg(format!("-j{}", conf.build_threads))
+        .arg(format!("install")), "make");
+}
+
+fn blob_backends(conf: &Config, build_dir: &std::path::PathBuf) -> (Vec<String>, Vec<String>) {
+    let mut backend_dirs :Vec<String>= Vec::new();
+    let mut backends :Vec<String> = Vec::new();
+
+    if conf.use_lib {
+        backend_dirs.push(conf.lib_dir.to_owned());
+    } else {
+        backend_dirs.push(build_dir.join("package/lib").to_str().to_owned().unwrap().to_string());
+    }
+
+    if conf.build_cuda == "ON" {
+        backends.push("afcuda".to_string());
+        backends.push("nvvm".to_string());
+        if cfg!(windows) {
+            backend_dirs.push(format!("{}\\lib\\x64", conf.cuda_sdk));
+            backend_dirs.push(format!("{}\\nvvm\\lib\\x64", conf.cuda_sdk));
+        } else {
+            backend_dirs.push(format!("{}/lib64", conf.cuda_sdk));
```

Comment:
This is not the case for OSX. The lib directory is {CUDA_ROOT}/lib
Can you add this to the build.conf as a parameter maybe?